src/storage/src/hummock/state_store.rs

```diff
@@ -186,42 +186,8 @@ impl StateStore for HummockStorage {
             }
             _ => return Ok(()),
         };
-        let mut receiver = self.version_update_notifier_tx.subscribe();
-        // avoid unnecessary check in the loop if the value does not change
-        let max_committed_epoch = *receiver.borrow_and_update();
-        if max_committed_epoch >= wait_epoch {
-            return Ok(());
-        }
-        loop {
-            match tokio::time::timeout(Duration::from_secs(30), receiver.changed()).await {
-                Err(elapsed) => {
-                    // The reason that we need to retry here is batch scan in
-                    // chain/rearrange_chain is waiting for an
-                    // uncommitted epoch carried by the CreateMV barrier, which
-                    // can take unbounded time to become committed and propagate
-                    // to the CN. We should consider removing the retry as well as wait_epoch
-                    // for chain/rearrange_chain if we enforce
-                    // chain/rearrange_chain to be scheduled on the same
-                    // CN with the same distribution as the upstream MV.
-                    // See #3845 for more details.
-                    tracing::warn!(
-                        "wait_epoch {:?} timeout when waiting for version update elapsed {:?}s",
-                        wait_epoch,
-                        elapsed
-                    );
-                    continue;
-                }
-                Ok(Err(_)) => {
-                    return Err(HummockError::wait_epoch("tx dropped").into());
-                }
-                Ok(Ok(_)) => {
-                    let max_committed_epoch = *receiver.borrow();
-                    if max_committed_epoch >= wait_epoch {
-                        return Ok(());
-                    }
-                }
-            }
-        }
+        let receiver = self.version_update_notifier_tx.subscribe();
+        wait_for_epoch(receiver, wait_epoch).await
```

Comment:
I think we can include the `notifier.subscribe()` in `wait_for_epoch`, and we pass the notifier as a parameter of `wait_for_epoch`.